plumbing/transport/common.go

```diff
@@ -131,7 +131,7 @@ type PushRequest struct {
 	Progress sideband.Progress
 
 	// Options is a set of push-options to be sent to the server during push.
-	Options map[string]string
+	Options []string
```

Comment:
As per upstream [docs](https://git-scm.com/docs/git-push#Documentation/git-push.txt-pushpushOption):

> This is a multi-valued variable, and an empty value can be used in a higher priority configuration file... to clear the values inherited from a lower priority configuration files

A map here would make that overwritting logic more straight-foward.

Reply:
This is only applicable when no `git push --push-option` are given.

In that case, options should be read from the repository and global config files. For example, to clear config options described in `/etc/gitconfig`, you would define new options in `.git/config` of the repository and use `push-options = ` to clear the global ones.

See https://git-scm.com/docs/git-push#Documentation/git-push.txt---push-optionltoptiongt

Which means when you use `git push --push-option` you can send duplicates and any string, see https://github.com/git/git/blob/485f5f863615e670fd97ae40af744e14072cfe18/send-pack.c#L375

Reply:
Fair point, thanks for clarifying.